### ISSUE TITLE
Lock scroll and zoom. Click to toggle unlock.

### ISSUE THREAD

**Issue by WardCunningham** [clemsos/wiki-plugin-map#18]
Authors select good first bounds for maps. Readers should not easily lose what the author intended. But readers and authors both will want to explore vicinity at will, one for context, the other to add more markers. Readers and authors will want again to view mechanically framed bounds, possibly modified bounds, and even alternate back and forth routinely.

We use click on a single page to "center" it in the lineup. I suggest we use the same gesture to unlock a map and then later "center" it back on its current bounds. The lock state should survive editing but be set back to locked on page reload.

I come to this opinion after hours of authoring with our newly functional map markup.

**Comment by opn** [comment]
I agree with the importance of stopping the zooming of maps simply by two-fingered scroll gestures of the mouse. Really hard to scroll down the page with a map on it with this active by default. 

However an alternative idea for the unlocking of the map would be to unlock the map after a user has used the + or - icons to zoom. Having show interest it is relatively safe to assume that further interaction with the map will be intended as zoom or pan.

Re-locking the map would be triggered by a page re-load or clicking outside the map (if the latter is feasible). The single-click toggle to lock and unlock is good - but a little too fragile for me. It emphasises the ability to interact with a map over the ability to scroll the page without getting unwanted effects. A user trying to scroll a page may start by clicking on the map. Since panning and zooming a page is not as important a feature as simply presenting the right map and enabling easy authoring of the map you wish to show on your site, I feel that making it a bit harder to pan-and zoom is justified.

**Comment by paul90** [comment]
Elsewhere, [fedwiki/wiki-plugin-map#13](https://github.com/fedwiki/wiki-plugin-map/pull/13), I looked at a fairly simplistic approach of disabling the map interaction when not editing it. Might be a useful starting point...
